src/core/atomic.d

```diff
@@ -190,6 +254,14 @@ else version( AsmX86_32 )
         //
         // +=   -=  *=  /=  %=  ^^= &=
         // |=   ^=  <<= >>= >>>=    ~=
+        static if( op == "+=" && __traits(isIntegral, T) && is(T==V1) ) {
```

Comment:
What does this code do:
asm { mov tmp, RAX; }
return cast(T)(tmp + mod);

I also didn't understand why this:
return oval + mod;

Why is any addition going on here (and why with mod)?